cmd/argocd/commands/app.go

```diff
@@ -445,71 +457,71 @@ func NewApplicationSetCommand(clientOpts *argocdclient.ClientOptions) *cobra.Com
 	return command
 }
 
-func setAppOptions(flags *pflag.FlagSet, app *argoappv1.Application, appOpts *appOptions) int {
+func setAppSpecOptions(flags *pflag.FlagSet, spec *argoappv1.ApplicationSpec, appOpts *appOptions) int {
```

Comment:
this method CANNOT set app fields (because the app is actually ignored by the invoker) and so I renamed it